server/src/main/java/org/elasticsearch/script/field/vectors/BinaryDenseVectorDocValuesField.java

```diff
@@ -54,16 +57,17 @@ public DenseVector get() {
         if (isEmpty()) {
             return DenseVector.EMPTY;
         }
-
-        return new BinaryDenseVector(value, dims, indexVersion);
+        VectorEncoderDecoder.decodeDenseVector(value, vectorValue);
+        return new BinaryDenseVector(vectorValue, value, dims, indexVersion);
     }
 
     @Override
     public DenseVector get(DenseVector defaultValue) {
         if (isEmpty()) {
             return defaultValue;
         }
-        return new BinaryDenseVector(value, dims, indexVersion);
+        VectorEncoderDecoder.decodeDenseVector(value, vectorValue);
```

Comment:
It's possible `get` is used w/ the same value twice. Wonder if it makes sense to cache it?

Reply:
I can cache it for sure.

Reply:
added code to cache